core/src/main/java/com/novoda/merlin/service/request/HttpRequestMaker.java

```diff
@@ -27,8 +29,8 @@ public Request head(String endpoint) {
         }
     }
 
-    private HttpURLConnection connectTo(String endpoint) throws IOException {
-        URL url = new URL(endpoint);
+    private HttpURLConnection connectTo(Endpoint endpoint) throws IOException {
+        URL url = new URL(endpoint.asString());
```

Comment:
Could just have `endpoint.asUrl()` no? this is the only place `asString` is called